fog/test-client/src/counters.rs

```diff
@@ -108,3 +116,76 @@ lazy_static::lazy_static! {
     /// This is updated after every transfer attempt.
     pub static ref LAST_POLLING_SUCCESSFUL: IntGauge = OP_COUNTERS.gauge("last_polling_successful");
 }
+
+/// Metrics for each individual client.
+/// Clients are identified by their index in the config.
+#[derive(Clone)]
+pub struct ClientMetrics {
+    balance_gauges: IntGaugeVec,
+    balance_block_count_gauges: IntGaugeVec,
+}
+
+impl Default for ClientMetrics {
+    fn default() -> Self {
+        ClientMetrics {
+            balance_gauges: IntGaugeVec::new(
+                Opts::new(
+                    "test_client_balance",
+                    "Balance of each individual test client",
+                ),
+                &["client_index", "token_id"],
+            )
+            .unwrap(),
+            balance_block_count_gauges: IntGaugeVec::new(
+                Opts::new(
+                    "test_client_balance_block_count",
+                    "Block count at which the balance was last updated",
+                ),
+                &["client_index"],
+            )
+            .unwrap(),
+        }
+    }
+}
+
+impl ClientMetrics {
+    /// Create a new ClientMetrics instance and register it with the Prometheus
+    /// crate.
+    pub fn new_and_registered() -> Self {
+        let client_metrics = Self::default();
+        prometheus::register(Box::new(client_metrics.clone()))
+            .expect("ClientMetrics registration on Prometheus failed.");
+        client_metrics
+    }
+
+    /// Update the balance and block count for a given client.
+    pub fn update_balance(
+        &self,
+        client_index: usize,
+        balances: &HashMap<TokenId, u64>,
+        block_count: impl Into<u64>,
+    ) {
+        let block_count = block_count.into();
+        for (token_id, balance) in balances {
+            self.balance_gauges
+                .with_label_values(&[&client_index.to_string(), &token_id.to_string()])
+                .set(*balance as i64);
+        }
+        self.balance_block_count_gauges
+            .with_label_values(&[&client_index.to_string()])
+            .set(block_count as i64);
+    }
+}
+
+impl Collector for ClientMetrics {
+    fn desc(&self) -> Vec<&Desc> {
+        self.balance_gauges.desc()
+    }
```

Comment:
Missing the bock count
```suggestion
    fn desc(&self) -> Vec<&Desc> {
        let mut metrics = Vec::with_capacity(2);
        metrics.extend(self.balance_gauges.desc());
        metrics.extend(self.balance_block_count_gauges.desc());
        metrics
    }
```

Reply:
Good catch!